### PR TITLE
envconsul 0.7.2 (new formula)

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
There is a lot of messing around going on because the envconsul documentation is very incomplete on how to build it, since everybody seems to use the mystery binaries that they provide.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by AndreasDahl** [comment]
Just putting this up here: https://github.com/hashicorp/envconsul

**Comment by Leonidas-from-XIV** [comment]
It works on my machine but fails on Jenkins, maybe @ilovezfs knows what the problem is, since he's maintaining a similar tool, consul-template.

**Comment by Leonidas-from-XIV** [comment]
Ha, I'm getting better at this. Instead of `go install` I `go build` now, which does not touch all kind of compiler stuff and instead just outputs a binary, just like I wanted.

**Comment by Leonidas-from-XIV** [comment]
Is there something wrong with it? `envconsul` has about 1k Github stars and it would be nice to be able to use a binary built from source managed by homebrew and receiving updates instead of their mystery binary that you have to maintain yourself.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/envconsul.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    ENV["GOPATH"] = gopath
+    cd envconsul_path do
+      system "go", "build"
+    end
+    bin.install envconsul_path/"envconsul"
+  end
+
+  test do
+    system "#{bin}/envconsul", "-version"
```
Comment (ilovezfs) [inline Formula/envconsul.rb]: This needs to be a test that exercises and verifies the functionality of the software.

File: Formula/envconsul.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class Envconsul < Formula
+  desc "Launch process with environment variables from Consul and Vault"
+  homepage "https://github.com/hashicorp/envconsul"
+  url "https://github.com/hashicorp/envconsul/archive/v0.7.2.tar.gz"
+  sha256 "c2b2723089f82f7b1623676fda8d378795bf87b4dbc6d4b297e5fc27aeab0aca"
+  depends_on "go" => :build
+
+  def install
+    gopath = buildpath/"go"
```
Comment (ilovezfs) [inline Formula/envconsul.rb]: Please take a look at some of the other formulae that depend on `go` to make this install method more idiomatic. For example, see `flint-checker` and `goose`.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/envconsul.rb (hunk 1 above)
Comment (Leonidas-from-XIV) [reply Formula/envconsul.rb]: I am not quite sure how to do that short of requiring the vault formula, setting up a vault server and testing against that. Is that the proper course of action?

File: Formula/envconsul.rb (hunk 2 above)
Comment (Leonidas-from-XIV) [reply Formula/envconsul.rb]: Done, made it much easier indeed. Thanks!

**Comment by Leonidas-from-XIV** [comment]
I have addressed most of the issues, the only thing I am not sure about is how to test the binary, since it requires a running vault to connect to.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@Leonidas-from-XIV you can probably do something like the `fabio` formula does in its test block.

**Comment by ilovezfs** [comment]
@Leonidas-from-XIV Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**